standalone/lcio2edm4hep.cpp

```diff
@@ -29,9 +30,29 @@ std::vector<std::pair<std::string, std::string>> getNamesAndTypes(const std::str
       return {};
     }
     std::string name = line.substr(0, delimiter_pos);
-    std::string type = line.substr(delimiter_pos + 1);
+    std::string type_space = line.substr(delimiter_pos + 1);
+    int first_letter;
+    for (auto elem :line.substr(delimiter_pos + 1)){
+      if (elem == ' '){ 
+        first_letter ++;
+      }
+      else{
+        continue;
+      }
+    }
+    std::string type line.substr(first_letter+delimiter_pos);
     names_types.emplace_back(name, type);
   }
+  */
+  while(std::getline(input_file, line)) {
+  std::stringstream sline(std::move(line));
+  std::string name, type;
+  if (!(sline >> name >> type && sline.eof())) {
+    std::cerr << "need a name and a type per line" << std::endl;
+    return{};
+  }
+  names_types.emplace_back(name, type);
```

Comment:
```suggestion
  names_types.emplace_back(std::move(name), std::move(type));
```
Just to avoid unnecessary string copies.